Sources/NIOInstrumentation/HTTPClientRequestPartWithBaggage.swift

```diff
@@ -0,0 +1,12 @@
+import BaggageContext
+import NIOHTTP1
+
+public struct HTTPClientRequestPartWithBaggage {
+    public let requestPart: HTTPClientRequestPart
+    public let baggage: BaggageContext
+
+    public init(requestPart: HTTPClientRequestPart, baggage: BaggageContext) {
+        self.requestPart = requestPart
+        self.baggage = baggage
+    }
+}
```

Comment:
TBH would put it in the same file as `BaggageContextOutboundHTTPHandler`, no strong opinion though.

WDYT?

Reply:
Sure, me neither. I'll add it to the file.